### PR TITLE
INT-1982 - UPS Pickup Retry

### PR DESCRIPTION
**Pull request by devin-haslam**

### CODE REVIEW

File: fedex.go [hunk 1]
```diff
@@ -46,68 +47,77 @@ func init() {
...
 
-func pickupTimeWindow(pickupAddress models.Address, numDaysToDelay int) (*models.PickupTimeWindow, error) {
+func pickupTimeWindow(clock clock.Clock, pickupAddress models.Address, pickupOffset *models.PickupOffset) (*models.PickupTimeWindow, error) {
 	location, err := toLocation(pickupAddress)
 	if err != nil {
 		location = laTimeZone
 	}
 
-	readyTime := time.Now().In(location).Add(time.Duration(numDaysToDelay*24) * time.Hour)
+	readyTime := time.Now().In(location).Add(time.Duration(pickupOffset.Days*24) * time.Hour)
```
Comment (jelott) [inline fedex.go]: Did you mean for this to be clock.Now()?

Comment (devin-haslam) [reply fedex.go]: yes, thank you. Updated!

File: fedex_test.go [hunk 2]
```diff
@@ -5,12 +5,15 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"os"
+	// "os"
```
Comment (jelott) [inline fedex_test.go]: Can we remove this vs. commenting out.

File: fedex_test.go [hunk 3]
```diff
@@ -723,6 +726,101 @@ func testShipSmartPostSuccess(t *testing.T, fedexAccount Fedex) {
...
+		assert.Equal(t, pickupTimeWindow.ReadyTime.Hour(), 10)
+		assert.Equal(t, pickupTimeWindow.ReadyTime.Minute(), 45)
+		assert.Equal(t, pickupTimeWindow.CloseTime.Hour(), 18)
+		assert.Equal(t, pickupTimeWindow.CloseTime.Minute(), 45)
+	})
+
+	t.Run("earlier pickup", func(t *testing.T) {
+		mockClock := clockMock.NewMockClock(ctrl)
+		testDate := time.Date(2023, 1, 2, 0, 0, 0, 0, time.UTC)
+		mockClock.EXPECT().Now().Return(testDate)
```
Comment (jelott) [inline fedex_test.go]: I think we should append .Times(1) to the end to ensure the clock is actually called.

Comment (devin-haslam) [reply fedex_test.go]: good call, updated

**Review by jelott — APPROVED** [review]

**Pull request merged**